src/util.h

```diff
@@ -84,6 +84,27 @@ static SECP256K1_INLINE void *checked_realloc(const secp256k1_callback* cb, void
     return ret;
 }
 
+#if defined(__BIGGEST_ALIGNMENT__)
+#define ALIGNMENT __BIGGEST_ALIGNMENT__
+#else
+/* Using 16 bytes alignment because common architectures never have alignment
+ * requirements above 8 for any of the types we care about. In addition we
+ * leave some room because currently we don't care about a few bytes. */
+#define ALIGNMENT 16
+#endif
+
+#define ROUND_TO_ALIGN(size) (((size + ALIGNMENT - 1) / ALIGNMENT) * ALIGNMENT)
+
+static SECP256K1_INLINE void *manual_alloc(void** prealloc_ptr, size_t alloc_size, void* base, size_t max_size) {
+    size_t aligned_alloc_size = ROUND_TO_ALIGN(alloc_size);
+    void* ret = *prealloc_ptr;
+    CHECK((char*)*prealloc_ptr != NULL);
```

Comment:
Todo: This check is not enough. The user could give us a NULL pointer, and then we want to call the callback and not just abort.

Reply:
I think it's fine. We call this only internally, so actually even a `VERIFY_CHECK` would be sufficient.

In the public-API-facing function where this pointer originates we should use an `ARG_CHECK`.